src/libraries/System.Private.CoreLib/src/System/Runtime/InteropServices/ComWrappers.cs

```diff
@@ -1337,16 +1336,11 @@ private sealed class RcwCache
                     // Return our target object.
                     return (wrapper, comProxy);
                 }
-                finally
-                {
-                    _lock.ExitWriteLock();
-                }
             }
 
             public object? FindProxyForComInstance(IntPtr comPointer)
             {
-                _lock.EnterReadLock();
-                try
+                lock (_lock)
```

Comment:
Here a dreadlock is not enough. It updates the hashtable under some circumstances, which may go concurrent under a read lock.

Reply:
Upgrading to a write lock before the update may help too... But the single lock is probably just as easy.